src/Reflection/ReflectionClass.php

```diff
@@ -350,7 +350,6 @@ function (ReflectionMethod $method) use ($filter) {
      */
     public function getImmediateMethods(?int $filter = null) : array
     {
-        /** @var \ReflectionMethod[] $methods */
```

Comment:
why removing this?

Reply:
Because it can be inferred easily

Reply:
`array_map()`'s documented return value is `array` - unless we have generics, this isn't inferred by tooling that doesn't assume implementation details of `array_map`, so the change should be reverted.

Reply:
Hmm, errata - it can be inferred, it's just that most tooling won't do recursive inferring of types, so this can't really be removed unless the ecosystem evolves

Reply:
![evolution](https://user-images.githubusercontent.com/2292638/35768738-f4fef8c0-08ce-11e8-8474-4ffa13cf43bc.jpg)

Reply:
I'll revert this

Reply:
Oh this was also inaccurate, which is why I made the change in the first place (it shouldn't have the leading slash)

Reply:
I like the pic :D

I know the tool you built is doing it, but the majority still needs to catch up ;-)